chatkit-core/src/test/kotlin/com/pusher/chatkit/cursors/CursorStoreApplyingReadStateUpdatedTest.kt

```diff
@@ -0,0 +1,149 @@
+                        )
+                ))
+            }
+
+
+            it("then the result indicates that nothing has been applied") {
+                assertThat(applyEventResult).isEmpty()
+            }
+            it("then the store contains the initial cursor") {
+                assertThat(subject[initialCursor.userId][initialCursor.roomId])
+                        .isEqualTo(initialCursor)
+            }
+        }
+        describe("when applying ReadStateUpdated with a new cursor") {
+            val cursor = Cursor("alice", "roomId1", position = 2)
+
+            lateinit var applyEventResult: List<UserInternalEvent>
+            beforeEachTest {
+                applyEventResult = subject.applyEvent(UserSubscriptionEvent.ReadStateUpdatedEvent(
+                        RoomReadStateApiType(
+                                roomId = "roomId1",
+                                unreadCount = 0,
+                                cursor = cursor
+                        )
+                ))
+            }
+
+            it("then the result indicates that the new cursor has been stored") {
+                assertThat(applyEventResult).containsExactly(UserInternalEvent.NewCursor(cursor))
+            }
+            it("then the store contains the new cursor") {
+                assertThat(subject[cursor.userId][cursor.roomId]).isEqualTo(cursor)
+            }
+        }
+        describe("when applying ReadStateUpdated for another room with no Cursor") {
+            lateinit var applyEventResult: List<UserInternalEvent>
+            beforeEachTest {
+                applyEventResult = subject.applyEvent(UserSubscriptionEvent.ReadStateUpdatedEvent(
+                        RoomReadStateApiType(
+                                roomId = "roomId1001",
+                                unreadCount = 1,
+                                cursor = null
+                        )
+                ))
+            }
+
+            it("then the result indicates that nothing has been applied") {
+                assertThat(applyEventResult).isEmpty()
+            }
+            it("then the store contains the initial cursor") {
+                assertThat(subject[initialCursor.userId][initialCursor.roomId])
+                        .isEqualTo(initialCursor)
+            }
+        }
+        describe("when applying ReadStateUpdated for another room with a Cursor") {
+            val cursor = Cursor("alice", "roomId1001", position = 2)
+
+            lateinit var applyEventResult: List<UserInternalEvent>
+            beforeEachTest {
+                applyEventResult = subject.applyEvent(UserSubscriptionEvent.ReadStateUpdatedEvent(
+                        RoomReadStateApiType(
+                                roomId = "roomId1001",
+                                unreadCount = 1,
+                                cursor = cursor
+                        )
+                ))
+            }
+
+
```

Comment:
looks like L134 and L135 are both empty too.

Reply:
sorry, not sure how it happened, probably I was too keen hitting enter to the rhythm of the music on my headphones ;) I will correct that, and switch back to my usual "relax nature" radio ;) cheers!